### ISSUE TITLE
Grouped/Sequenced Canary Promotions

### ISSUE THREAD

**Issue by stevenheinze57** [hashicorp/nomad#11822]
Is it possible to include as a feature, grouped/sequenced canary promotions? To better explain this I'll give the below example: 

Running allocations: 10 
Canary Allocations: 2 
I'd like to update my 10 allocations 2 at a time in this scenario, which would total to 5 promotions. When all 5 promotions are successful then all 10 allocations would have been updated. 

I actually was originally under the impression that this was the default behavior until I started attempting to use canary updates with my deployments. The behavior I'm seeing is that promoting the 2 canaries will then force a rolling update across the other 8 allocations. In my case I have a nosql deployment I am managing and need the ability to run configuration updates prior to promotions. This give me fine grained control of the update process and reserves my ability to rollback if needed.

**Comment by tgross** [comment]
Hi @stevenheinze57! Just in case I'm misreading the scenario, there's already a `max_parallel` field that makes sure that deployments go out `n` at a time. So you could do something like this:

```hcl
update {
  max_parallel      = 2
  health_check      = "checks"
  min_healthy_time  = "5m"
  healthy_deadline  = "10m"
  progress_deadline = "20m"
  auto_revert       = true
  auto_promote      = false
  canary            = 2
  stagger           = "1m"
}
```

This would result in 2 canaries, which you would manually promote. And then once those are promoted Nomad would deploy at most 2 at a time, waiting 5min (`min_healthy_time`) before moving on to the next allocation.

But it sounds like you might be looking to manually promote _all_ allocation updates?

**Comment by stevenheinze57** [comment]
@tgross  You are correct. I'd like to manually promote at most, 2 allocations at a time, instead of having the max parallel handle that process automatically. 

promote 0 and 1
promote 2 and 3 
promote 4 and 5 
promote 6 and 7 
promote 8 and 9 
etc....

**Comment by tgross** [comment]
I've renamed this issue and tagged it to get some attention for roadmapping and further discussion.
